agogosml_cli/cli/generate.py

```diff
@@ -139,11 +136,22 @@ def extract_azure_template_vars(manifest: dict) -> dict:
         raise click.Abort()
     else:
         acr = azure_props['azureContainerRegistry']
-        template_vars['AZURE_CONTAINER_REGISTRY'] = acr
-        if not acr.endswith('/'):
-            acr += '/'
-        template_vars['AZURE_DOCKER_BUILDARGS'] = \
-            '--build-arg CONTAINER_REG=$(container_registry) --build-arg AGOGOSML_TAG=$(Build.BuildId)'
+        try:
+            extract_result = tldextract.extract(acr)
+        except ValueError:
+            # Handle situation where template folder exists and force is not set to true
+            click.echo('Azure Container Registry property is not parseable as a url. \
+                Valid property: nameacr.azurecr.io.')
+            raise click.Abort()
+
+        if extract_result.registered_domain != "azurecr.io":
```

Comment:
won't this block us from using a non-azure registry like docker hub or on-premises?

Reply:
Acknowledged. I am going to promote this setting to a global setting and restrict this additional check to only azure settings. I added a user story to the backlog.